src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.cs

```diff
@@ -2129,6 +2129,43 @@ internal static bool IsCustomCultureId(int cultureId)
             return cultureId == CultureInfo.LOCALE_CUSTOM_DEFAULT || cultureId == CultureInfo.LOCALE_CUSTOM_UNSPECIFIED;
         }
 
+        private string[] GetNativeDigits()
+        {
+            string[] result = NumberFormatInfo.s_asciiDigits;
+
+            // LOCALE_SNATIVEDIGITS (array of 10 single character strings).
+            string digits = GetLocaleInfoCoreUserOverride(LocaleStringData.Digits);
+
+            // if digits.Length < NumberFormatInfo.s_asciiDigits.Length means the native digits setting is messed up in the host machine.
+            // Instead of throwing IndexOutOfRangeException that will be hard to diagnose after the fact, we'll fall back to use the ASCII digits instead.
```

Comment:
would this method be clearer if it did early out (ie `return result;`) in various places? 

eg 
if (digits.Length < blahblah)
{
    return result;
}

and again in the loop? I find early out easier to read from top to bottom but it may be me.

Reply:
Sounds good.